pettingzoo/test/max_cycles_test.py

```diff
@@ -6,21 +6,27 @@ def max_cycles_test(mod):
     parallel_env = mod.parallel_env(max_cycles=max_cycles)
 
     observations = parallel_env.reset()
-    dones = {agent: False for agent in parallel_env.agents}
+    terminations = {agent: False for agent in parallel_env.agents}
+    truncations = {agent: False for agent in parallel_env.agents}
     test_cycles = (
         max_cycles + 10
     )  # allows environment to do more than max_cycles if it so wishes
     for step in range(test_cycles):
         actions = {
             agent: parallel_env.action_space(agent).sample()
             for agent in parallel_env.agents
-            if not dones[agent]
+            if not {
+                x[0]: x[1] or y[1]
+                for x, y in zip(terminations.items(), truncations.items())
+            }[agent]
```

Comment:
I apologize for being overly pedantic, but would 
```py
if not (terminations[agent] or truncations[agent]):
  ...
```
not work instead? Seems like it would be a faster solution.

Reply:
No need to apologise! Good spot, I was tunnelvisioning on list comprehensions.